### PR TITLE
[zh-cn] sync blog 2023-07-06-confidential-kubernetes

### PR DESCRIPTION
**Pull request by asa3311**
link #41971

### CODE REVIEW

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,687 @@
+---
+layout: blog
+title: "Kubernetes 机密：使用机密虚拟机和安全区来增强你的集群安全性。"
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:3]:
```suggestion
title: "Kubernetes 机密：使用机密虚拟机和安全区来增强你的集群安全性"
```

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+by introducing a hardware-enforced Trusted Execution Environment (TEE).
+-->
+在这篇博客文章中，我们将介绍机密计算（Confidential Computing，简称 CC）的概念，
+以增强任何计算环境的安全和隐私属性。此外，我们将展示云原生生态系统，
+特别是 Kubernetes，如何从新的计算范式中受益。
+
+机密计算是一个先前在云原生领域中引入的概念。
+[机密计算联盟](https://confidentialcomputing.io/)(Confidential Computing Consortium，简称 CCC) 
+是 Linux 基金会中的一个项目社区，
+已致力于[定义和启用机密计算](https://confidentialcomputing.io/wp-content/uploads/sites/85/2019/12/CCC_Overview.pdf)。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:49]:
```suggestion
致力于[定义和启用机密计算](https://confidentialcomputing.io/wp-content/uploads/sites/85/2019/12/CCC_Overview.pdf)。
```

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+在他们的[白皮书](https://confidentialcomputing.io/wp-content/uploads/sites/85/2023/01/CCC-A-Technical-Analysis-of-Confidential-Computing-v1.3_Updated_November_2022.pdf)中，
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:50]:
```suggestion
在[白皮书](https://confidentialcomputing.io/wp-content/uploads/sites/85/2023/01/CCC-A-Technical-Analysis-of-Confidential-Computing-v1.3_Updated_November_2022.pdf)中，
```

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,687 @@
...
+他们为使用机密计算提供了很好的动机。
+
+   > 数据存在于三种状态：传输中、静态存储和使用中。...保护所有状态下的敏感数据比以往任何时候都更加关键。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:53]:
```suggestion
   > 数据存在于三种状态：传输中、静态存储和使用中。保护所有状态下的敏感数据比以往任何时候都更加关键。
```

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,687 @@
...
+
+<!--
+[IBM Secure Execution for Linux](https://www.ibm.com/docs/en/linux-on-systems?topic=virtualization-secure-execution)
+lets you run your Kubernetes cluster's nodes as KVM guests within a trusted execution environment on
+IBM Z series hardware. You can use this hardware-enhanced virtual machine isolation to
+provide strong isolation between tenants in a cluster, with hardware attestation about the (virtual) node's integrity.
+-->
+[IBM Secure Execution for Linux](https://www.ibm.com/docs/en/linux-on-systems?topic=virtualization-secure-execution)
+允许你在 IBM Z 系列硬件的可信执行环境内以 KVM 客户端的形式运行 Kubernetes 集群的节点。
+你可以使用这种硬件增强的虚拟机隔离在集群中的租户之间提供强大的隔离，
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:138]:
```suggestion
你可以使用这种硬件增强的虚拟机隔离机制为集群中的租户之间提供稳固的隔离，
```

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,687 @@
...
+### Security properties and feature set
+
+In the following sections, we will review the security properties and additional features
+these new technologies bring to the table. Only some solutions will provide all properties;
+we will discuss each technology in further detail in their respective section.
+-->
+### 安全属性和特性功能  {#security-properties-and-feature-set}
+
+在接下来的部分，我们将回顾这些新技术带来的安全属性和额外功能。
+只有部分解决方案会提供所有属性；我们将在各自的部分中更详细地讨论每项技术。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:151]:
```suggestion
下文将回顾这些新技术所带来的安全属性和额外功能。
只有部分解决方案会提供所有属性；我们将在各自的小节中更详细地讨论每项技术。
```

**Review by windsonsea — COMMENTED** [review]
See preview:
https://deploy-preview-41806--kubernetes-io-main-staging.netlify.app/zh-cn/blog/2023/07/06/confidential-kubernetes/

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,687 @@
...
+in use in the TEE. This provides us with the highly desired feature to secure
+**data in use**. Depending on the specific TEE used, both code and data may be protected
+from outside viewers. The differences in TEE architectures and how their use
+in a cloud native context are important considerations when designing end-to-end security
+for sensitive workloads with a minimal **Trusted Computing Base** (TCB) in mind. CCC has recently
+worked on a [common vocabulary and supporting material](https://confidentialcomputing.io/wp-content/uploads/sites/85/2023/01/Common-Terminology-for-Confidential-Computing.pdf)
+that helps to explain where confidentiality boundaries are drawn with the different TEE
+architectures and how that impacts the TCB size.
+-->
+**机密性**属性确保在使用 TEE 时信息无法被查看。这为我们提供了保护**使用中的数据**的高度期望的功能。
```
Comment (mengjiao-liu) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:164]:
```suggestion
**机密性**属性确保在使用 TEE 时信息无法被查看。这为我们提供了非常需要的的功能以保护**使用中的数据**。
```

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,687 @@
...
+密钥存储在 CPU 内部，无法访问。
+
+<!--
+Since the enclaves are process isolated, the operating system's libraries are not usable as is;
+therefore, SGX enclave SDKs are required to compile programs for SGX. This also implies applications
+need to be designed and implemented to consider the trusted/untrusted isolation boundaries.
+On the other hand, applications get built with very minimal TCB.
+-->
+由于 Enclave 是进程隔离的，操作系统的库不能直接使用；
+因此，需要 SGX Enclave SDK来编译针对 SGX 的程序。
```
Comment (mengjiao-liu) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:346]:
```suggestion
因此，需要 SGX Enclave SDK 来编译针对 SGX 的程序。
```

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,687 @@
...
+[Alibaba Cloud provides](https://www.alibabacloud.com/help/en/elastic-compute-service/latest/build-a-tdx-confidential-computing-environment)
+invitational preview instances, and
+[Azure has announced](https://techcommunity.microsoft.com/t5/azure-confidential-computing/preview-introducing-dcesv5-and-ecesv5-series-confidential-vms/ba-p/3800718)
+its TDX preview opportunity.
+-->
+与 SEV-SNP 一样，对 TDX 的客户端支持已经在
+[Linux Kernel 5.19版本中合并](https://www.phoronix.com/news/Intel-TDX-For-Linux-5.19)。
+然而，硬件支持将在 2023 年与 [Sapphire Rapids](https://en.wikipedia.org/wiki/Sapphire_Rapids) 一同发布：
+[阿里云提供](https://www.alibabacloud.com/help/en/elastic-compute-service/latest/build-a-tdx-confidential-computing-environment)
+邀请预览实例，同时，[Azure已经宣布](https://techcommunity.microsoft.com/t5/azure-confidential-computing/preview-introducing-dcesv5-and-ecesv5-series-confidential-vms/ba-p/3800718)
```
Comment (mengjiao-liu) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:420]:
```suggestion
邀请预览实例，同时，[Azure 已经宣布](https://techcommunity.microsoft.com/t5/azure-confidential-computing/preview-introducing-dcesv5-and-ecesv5-series-confidential-vms/ba-p/3800718)
```

File: content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,687 @@
...
+computing hardware in the form of [Hardware Security Modules](https://en.wikipedia.org/wiki/Hardware_security_module)
+(HSMs) and [Trusted Platform Modules](https://www.iso.org/standard/50970.html) (TPMs). These
+technologies provide trusted environments for shielded computations. They can
+store highly sensitive cryptographic keys and carry out critical cryptographic operations
+such as signing or encrypting data.
+-->
+## 可信执行环境  {#trusted-execution-environments}
+
+在过去的十多年里，可信执行环境（Trusted Execution Environments，简称 TEEs）
+以[硬件安全模块](https://en.wikipedia.org/wiki/Hardware_security_module)（Hardware Security Modules，简称 HSMs）
```
Comment (ydFu) [inline content/zh-cn/blog/_posts/2023-07-06-confidential-kubernetes.md:72]:
```suggestion
以[硬件安全模块](https://zh.wikipedia.org/zh-cn/%E7%A1%AC%E4%BB%B6%E5%AE%89%E5%85%A8%E6%A8%A1%E5%9D%97)（Hardware Security Modules，简称 HSMs）
```

**Review by ydFu — APPROVED** [review]
/lgtm

This article introduces the concept of Confidential Computing.

**Pull request merged**